pbparam/optimisation_problems/data_fit.py

```diff
@@ -65,8 +65,15 @@ def objective_function_full(opt_problem, x):
     y_sim = [solution[v](data["Time [s]"]) for v in variables_optimise]
     y_data = [data[v] for v in variables_optimise]
     sd = [x[opt_problem.map_inputs[k]] for k in opt_problem.cost_function_parameters]
+    weights = getattr(opt_problem, "weights", {}).get(
+        variables_optimise[0], [1 / np.nanmean(y_data)] * len(y_data)
+    )
+    if len(weights) == 1:
+        weights *= len(y_data)
+    elif len(weights) != len(y_data):
+        raise ValueError("Weights should have the same length as y_data.")
```

Comment:
Given that weights will not change during the optimisation problem, it would be better to do this check at the `__init__` method, so we only perform this once. This will also fix the failing tests due to the logging level.